tools/mason/MasonSearch.chpl

```diff
@@ -0,0 +1,78 @@
+
+use MasonHelp;
+use MasonUpdate;
+
+use FileSystem;
+use Regexp;
+
+//
+// TODO:
+// - order results by some kind of 'best match' metric
+// - allow regex searches
+// - allow for exclusion of a pattern
+//
+
+proc masonSearch(origArgs : [] string) {
+  var args : [1..origArgs.size] string = origArgs;
+
+  const printHelp = || reduce [a in args] (a == "-h" || a == "--help");
+  if printHelp {
+    masonSearchHelp();
+    exit(0);
+  }
+
+  updateRegistry("", args);
+
+  consumeArgs(args);
+
+  // If no query is provided, list all packages in registry
+  const query = if args.size > 0 then args.tail().toLower()
```

Comment:
What are your thoughts on making the default search behavior follow *smart-casing*?

if `query.isUpper()` then `ignorecase=false`, otherwise follow the same logic you have.

Reply:
I'd prefer to keep it simple for this first pass.

Reply:
Works for me - we should make it clear that `mason search` is case insensitive in the help output, if possible.